src/components/ModalWithConfirm.js

```diff
@@ -0,0 +1,38 @@
+import Modal from "./Modal.js";
+
+class ModalWithConfirm extends Modal {
+  constructor({ modalSelector, handleConfirmAction }) {
+    //
+    super(modalSelector);
+    this._modalSelector = document.querySelector(modalSelector);
+    this._modalConfirmBox = this._modalSelector.querySelector(
+      ".modal__confirm-box"
+    );
+    this._handleConfirmAction = handleConfirmAction;
+    this.isSubmitted = false;
+  }
+  open(currentCard, cardId) {
```

Comment:
Needs correction: `PopupWithConfirmation` is a general class. It can be used for any confirmation, not only card delete confirmation. This class should have a setter for `_handleFormSubmit`, not for card, so you can use a card in that handler in `index.js`.

```js
constructor(popupSelector) {
  super(popupSelector);
  this._form = this._modal.querySelector(".modal__form");
}

setAction(action) { 
  this._handleFormSubmit = action; 
} 

this._form.addEventListener("submit", (evt) => { 
  evt.preventDefault(); 
  this._handleFormSubmit(); 
})
```